src/murmur/Server.cpp

```diff
@@ -1171,114 +1171,128 @@ void Server::processMsg(ServerUser *u, Mumble::Protocol::AudioData audioData, Au
+									   m_channelListenerManager.getListenerVolumeAdjustment(pDst->uiSession, c->iId));
+				}
 			}
-		}
 
-		// Send audio to all users in the same channel
-		for (User *p : c->qlUsers) {
-			ServerUser *pDst = static_cast< ServerUser * >(p);
+			// Send audio to all users in the same channel
+			for (User *p : c->qlUsers) {
+				ServerUser *pDst = static_cast< ServerUser * >(p);
 
-			buffer.addReceiver(*u, *pDst, Mumble::Protocol::AudioContext::NORMAL, audioData.containsPositionalData);
-		}
+				buffer.addReceiver(*u, *pDst, Mumble::Protocol::AudioContext::NORMAL, audioData.containsPositionalData);
+			}
+
+			// Send audio to all linked channels the user has speak-permission
+			if (!c->qhLinks.isEmpty()) {
+				QSet< Channel * > chans = c->allLinks();
+				chans.remove(c);
 
-		// Send audio to all linked channels the user has speak-permission
-		if (!c->qhLinks.isEmpty()) {
-			QSet< Channel * > chans = c->allLinks();
-			chans.remove(c);
-
-			QMutexLocker qml(&qmCache);
-
-			for (Channel *l : chans) {
-				if (ChanACL::hasPermission(u, l, ChanACL::Speak, &acCache)) {
-					// Send the audio stream to all users that are listening to the linked channel
-					for (unsigned int currentSession : m_channelListenerManager.getListenersForChannel(l->iId)) {
-						ServerUser *pDst = static_cast< ServerUser * >(qhUsers.value(currentSession));
-						if (pDst) {
-							buffer.addReceiver(
-								*u, *pDst, Mumble::Protocol::AudioContext::LISTEN, audioData.containsPositionalData,
-								m_channelListenerManager.getListenerVolumeAdjustment(pDst->uiSession, l->iId));
+				QMutexLocker qml(&qmCache);
+
+				for (Channel *l : chans) {
+					if (ChanACL::hasPermission(u, l, ChanACL::Speak, &acCache)) {
+						// Send the audio stream to all users that are listening to the linked channel
+						for (unsigned int currentSession : m_channelListenerManager.getListenersForChannel(l->iId)) {
+							ServerUser *pDst = static_cast< ServerUser * >(qhUsers.value(currentSession));
+							if (pDst) {
+								buffer.addReceiver(
+									*u, *pDst, Mumble::Protocol::AudioContext::LISTEN, audioData.containsPositionalData,
+									m_channelListenerManager.getListenerVolumeAdjustment(pDst->uiSession, l->iId));
+							}
 						}
-					}
 
-					// Send audio to users in the linked channel
-					for (User *p : l->qlUsers) {
-						ServerUser *pDst = static_cast< ServerUser * >(p);
+						// Send audio to users in the linked channel
+						for (User *p : l->qlUsers) {
+							ServerUser *pDst = static_cast< ServerUser * >(p);
 
-						buffer.addReceiver(*u, *pDst, Mumble::Protocol::AudioContext::NORMAL,
-										   audioData.containsPositionalData);
+							buffer.addReceiver(*u, *pDst, Mumble::Protocol::AudioContext::NORMAL,
+											   audioData.containsPositionalData);
+						}
 					}
 				}
 			}
+
+			break;
 		}
-	} else if (u->qmTargets.contains(static_cast< int >(audioData.targetOrContext))) { // Whisper/Shout
```

Comment:
What happens when clients implementing this feature are connected to old servers? When they are using "server (send to others)" their packets will be dropped entirely, correct? Or will they be whisper/shout to some channel or user?

Reply:
They are requesting the audio to be sent to a non-registered whisper-target, which should lead to the packet just being dropped.

Reply:
The new loopback mode should probably only be used/offered to the user whilst connected to a server that actually supports this.

Reply:
I was thinking about only showing the option in the combobox when available (i.e. the server supports it), but the idea implies that:

1. It would not appear when not connected to a server.
2. It would be confusing as the user may think their client version doesn't provide the feature.

I think we should show a warning in the log box explaining that the selected loopback method is not supported due to the server not implementing it.

Reply:
I think hiding the option if the server doesn't support it makes sense. Only issue would be that it'd be confusing for users who configure loopback while disconnected and once they connect their chosen option is gone.

Reply:
I am not sure hiding would be a good UX. Either change the text of the combobox entries to add ``(unsupported by server)`` or use the log

Reply:
The log will be overlooked in like 90% of cases.

Reply:
Agreed, but hiding will create us a dozen bug reports xD

Reply:
True :)
We could make use of the mute-cue in cases where the user has selected loopback+broadcast but the server doesn't support it - because as far as other clients are concerned, you are muted (more or less). So maybe mute-cue + indicating in the settings window that the server doesn't support the selected loopback+broadcast would be an idea?